RecipeMatch/View Controllers/DetailsViewController.m

```diff
@@ -18,124 +18,107 @@ @interface DetailsViewController ()
          forControlEvents:UIControlEventTouchUpInside];
     [self.saveBtn addTarget:self action:@selector(didTapSave:)
          forControlEvents:UIControlEventTouchUpInside];
     
-    // setup like button
-    [APIManager checkIfLikedWithId:self.savedRecipe.recipeId andCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+    // setups like button
+    [APIManager checkIfRecipeIsAlreadyLikedWithId:self.savedRecipe.recipeId andCompletion:^(BOOL succeeded, NSError * _Nullable error) {
         if(succeeded == YES){
             [self.likeBtn setImage:[UIImage systemImageNamed:@"heart.fill"] forState:UIControlStateNormal];
+            self.liked = YES;
         } else{
             [self.likeBtn setImage:[UIImage systemImageNamed:@"heart"] forState:UIControlStateNormal];
+            self.liked = NO;
         }
     }];
 
-    //setup save button
-    [APIManager checkIfSavedWithId:self.savedRecipe.recipeId andCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+    //setups save button
+    [APIManager checkIfRecipeIsAlreadySavedWithId:self.savedRecipe.recipeId andCompletion:^(BOOL succeeded, NSError * _Nullable error) {
         if(succeeded == YES){
             [self.saveBtn setImage:[UIImage systemImageNamed:@"bookmark.fill"] forState:UIControlStateNormal];
+            self.saved = YES;
         } else{
             [self.saveBtn setImage:[UIImage systemImageNamed:@"bookmark"] forState:UIControlStateNormal];
+            self.saved = NO;
         }
     }];
 }
 
--(void)fetchRecipeInfo{
+// gets recipe details from recipe API
+- (void)fetchRecipeInfo{
     [[APIManager shared] getRecipeWithId:self.savedRecipe.recipeId andCompletion: ^(NSDictionary *recipe, NSError *error){
-        if(recipe)
-        {
+        if(recipe){
             self.fullRecipe = recipe;
             self.recipeTitle.text = recipe[@"label"];
             [self.source setTitle:recipe[@"source"] forState:UIControlStateNormal];
             [self.source.titleLabel setFont:[UIFont boldSystemFontOfSize:15]];
-           
             [self.source addTarget:self action:@selector(didTapSource:) forControlEvents:UIControlEventTouchUpInside];
-            
             self.recipeUrl = recipe[@"url"];
-          
             NSArray *ingrArray = recipe[@"ingredientLines"];
             NSString *ingrString = (NSString *)[ingrArray componentsJoinedByString:@"\r\r• "];
             self.ingredients.text = [@"• " stringByAppendingString:ingrString];
             self.yield.text = [NSString stringWithFormat:@"%@", recipe[@"yield"]];
-            
             NSString *imageUrl = recipe[@"image"];
             NSData * imageData = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString: imageUrl]];
             self.recipeImage.image = [UIImage imageWithData: imageData];
-            
             [self.view setNeedsDisplay];
-        } else {
-            NSLog(@"😫😫😫 Error getting recipe info: %@", error.localizedDescription);
```

Comment:
might be worth supporting this case in the future.
you can add a TODO remark (ie `//TODO: Add failure support`)